packages/components/src/molecules/Carousel/CarouselItem.tsx

```diff
@@ -18,19 +19,26 @@ function CarouselItem({
   totalItems,
   infiniteMode,
   isScrollCarousel,
+  marginRightValue,
 }: PropsWithChildren<CarouselItemProps>) {
   const { isItemVisible, shouldRenderItem } = useSlideVisibility({
     totalItems,
     currentSlide: state.currentItem,
     itemsPerPage: state.itemsPerPage,
   })
 
-  const style =
-    ((!isScrollCarousel && { width: '100%' }) as CSSProperties) ||
-    ((isScrollCarousel && {
-      maxWidth: '60%',
-      display: 'inline-block',
-    }) as CSSProperties)
+  const defaultStyle = { width: '100%' } as CSSProperties
+  const scrollCarouselStyle = {
+    width: `calc((100% - ${marginRightValue} * ${state.itemsPerPage - 1}) / ${
+      state.itemsPerPage
+    })`,
+  } as CSSProperties
+
+  const style = isScrollCarousel
+    ? state.itemsPerPage > 1
+      ? { ...scrollCarouselStyle }
+      : { ...defaultStyle }
+    : { ...defaultStyle }
```

Comment:
To be a little more readable, what do you think?
```suggestion
  const style = isScrollCarousel && state.itemsPerPage > 1
    ? { ...scrollCarouselStyle }
    : { ...defaultStyle }
```